### PR TITLE
heap_sort.cpp: Adds Heap Sort Algorithm

### PR DESCRIPTION
**Pull request by thebrahminator**
This adds Heap Sort Algorithm which sorts the elements
in ascending order.

Closes https://github.com/NITSkmOS/Algorithms/issues/22

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

Fixes #{ISSUE_NUMBER}

**By submitting this pull request I confirm I've read and complied with the
below declarations.**

- [x] I have read the [Contribution guidelines](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [commit guidelines](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md#write-good-commit-messages) for this project.
- [x] My code follows the [standard code structure](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md#samples-code).
- [x] This pull request has a descriptive title. For example, `{Tag}: Add {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.
- [x] This pull request shall only be reviewed and merged once all the checks pass. No maintainer or supporter shall be obliged to review it before this condition is met.
- [x] I have mentioned the issue number correctly (with hyperlink) in this pull request description.

After you submit your pull request, **DO NOT** click the 'Update Branch' button.

### CODE REVIEW

File: heap_sort/C++/heap_sort.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        cin>>array[i];
+    }
+    heapSort(array, n); 
+    cout << "Sorted array is \n"; 
+    for(int i=0; i<n; i++){
+        cout<<array[i]<<" ";
+    }
+    cout<<endl;
+    return 1;
+}
```
Comment (sangamcse) [inline heap_sort/C++/heap_sort.cpp]: EOL missing

File: heap_sort/C++/heap_sort.cpp [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+void heapSort(int input[], int n) 
+{ 
+    for (int i = n / 2 - 1; i >= 0; i--) 
+        heapify(input, n, i); 
+    for (int i=n-1; i>=0; i--) 
+    { 
+        swap(input[0], input[i]); 
+        heapify(input, i, 0); 
+    } 
+}
```
Comment (sangamcse) [inline heap_sort/C++/heap_sort.cpp]: blank line missing

**Comment by sangamcse** [comment]
You have changed from 3 commits to 6 commits. It will break my repo.

**Comment by thebrahminator** [comment]
Hey.

I'll try and reduce it. Sorry for this.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Pull request closed without merging**